qiskit_experiments/test/iq_pulse_backend.py

```diff
@@ -287,25 +313,42 @@ def run(self, run_input: Union[QuantumCircuit, List[QuantumCircuit]], **run_opti
         if isinstance(run_input, QuantumCircuit):
             run_input = [run_input]
 
-        experiment_unitaries = copy.deepcopy(self.default_pulse_unitaries)
-
         for circuit in run_input:
+            unitaries = {}
```

Comment:
Why are we moving this inside the circuits loop? in `FineAmp` for example, the same unitary is used for all circuits.

Reply:
Because different circuits may override the gates in different ways (e.g. Rabi arguably with different parameter values). FineAmp is a special case because the schedule is common to all circuits. Recall that `calibrations` is a variable that is local to each circuit.

Reply:
Right. I think i understand.

Reply:
Also, now in this current structure, we no longer need the default pulse unitaries since the transpilation adds calibrations for all supported gates. Right?